src/gadgets/ecc.rs

```diff
@@ -160,6 +160,74 @@ impl<C: CurveAffine<Base = F>, F: PrimeFieldBits, const T: usize> EccChip<C, F,
         Ok(acc)
     }
 
+    // optimization version for scalar_mul
+    // we assume the point p0 is not infinity here
+    pub fn scalar_mul_opt(
+        &self,
+        ctx: &mut RegionCtx<'_, C::Base>,
+        p0: &AssignedPoint<C>,
```

Comment:
`multiplicand`

Reply:
what do you mean by `multiplicand` here?

Reply:
I mean, rename `p0`

Reply:
multiplicand is not quite accurate here. To emphasize it is a point on the curve, what about `p0->pt`?

Reply:
Any mathematical terms in code are always a compromise, so if possible we would like to use names more familiar to programmers. 

The fact that it is a point reflects the type of this variable. 

There is a general convention that multipliers are usually denoted by lhs & rhs. What do you think?

Reply:
`lhs` and `rhs` are too general.  There is not confusion about it: It's multiplication between a point `pt` and a scalar `scalar_bits`. Very accurate.

Reply:
`pt: AssignedPoint` is to convey the same information twice, although it's actually just as important here to realize that this operand and that one is being multiplied by the other. However, this is not so fundamental

Reply:
Yeah, this is NBD. Let's just keep the initial name `p0` meaning the initial point.